### PR TITLE
Fix AV if allocation of error_message fails

### PR DESCRIPTION
**Pull request by Alan-Jowett**
Resolves: #3213 

## Description

Fix AV if allocation of error_message string fails.

## Testing

CI/CD

## Documentation

No.

## Installation

No.

### CODE REVIEW

File: libs/api/ebpf_api.cpp [hunk 1]
```diff
@@ -2829,7 +2829,7 @@ _Requires_lock_not_held_(_ebpf_state_mutex) _Must_inspect_result_ ebpf_result_t
         std::string log_message = "ebpf_object_open: error loading file:";
         log_message += path;
         log_message += ", error message:";
-        log_message += *error_message;
+        log_message += *error_message != NULL ? *error_message : "none";
```
Comment (dv-msft) [inline libs/api/ebpf_api.cpp:2832]: Might be more helpful to say something like "peparse error message dropped" instead of "none".

Comment (Alan-Jowett) [reply libs/api/ebpf_api.cpp:2832]: The source of the error message is not known at this point.

**Review by dv-msft — APPROVED** [review]

**Review by rectified95 — APPROVED** [review]

File: libs/api/ebpf_api.cpp [hunk 2]
```diff
@@ -2744,7 +2744,7 @@ _ebpf_enumerate_native_sections(
         std::scoped_lock pe_parse_lock(_pe_parse_mutex);
```
Comment (gtrevi) [inline libs/api/ebpf_api.cpp:2744]: Is it intentional throwing an exception if  `error_message` is null?

Comment (Alan-Jowett) [reply libs/api/ebpf_api.cpp:2744]: I don't understand the question. Constructing a std::string from a nullptr is undefined behavior, which triggers an AV on Windows. Hope that helps.

**Pull request merged**